portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,28 +27,28 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    String json = convertToJson(generateJsonAsArrayList());
-
-    response.setContentType("text/html;");
-    response.getWriter().println(json);
-  }
-
-  private ArrayList<String> generateJsonAsArrayList(){
-    ArrayList<String> msg = new ArrayList<String>(); 
-    msg.add("hello");
-    msg.add("world");
-    msg.add("Chen");
-
-    return msg;
-  }
-
-  private String convertToJson(ArrayList arraylist) {
-    Gson gson = new Gson();
-    String json = gson.toJson(arraylist);
-    return json;
-  }
-
+    private final CommentList mCommentList = new CommentList();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        String json = new Gson().toJson(mCommentList);
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String comment =  request.getParameter("user-comment");
+        if (comment.length() == 0) {
+            response.setContentType("text/html");
+            response.getWriter().println("Your comment is empty!");
+            return;
+        }
```

Comment:
Really nice job handling an edge case. When we start writing unit tests, this is a good example of something to test.